cranelift/codegen/src/isa/aarch64/abi.rs

```diff
@@ -927,10 +974,20 @@ impl ABICall for AArch64ABICall {
         adjust_stack(self.sig.stack_arg_space as u64, /* is_sub = */ false)
     }
 
-    fn gen_copy_reg_to_arg(&self, idx: usize, from_reg: Reg) -> Inst {
+    fn gen_copy_reg_to_arg(
+        &self,
+        idx: usize,
+        from_reg: Reg,
+        tmp1: Writable<Reg>,
```

Comment:
Hmm. I don't want to block this PR on it, but I'm thinking we should really clean up the ABI traits to take a `LowerCtx` and just emit instructions and allocate temps as needed directly, as you suggested above. @alexcrichton hit a similar issue with is stack-check PR (#1573). It's my design flaw originally (sorry!) so I can take this as a followup item once that PR and this one land.

Reply:
I looked into this, and I could make it work as discussed, but it puts the type parameter down to the implementation of `store_stack_sp`, which is unfortunate. Maybe we could revisit this later, and instead manually monomorphize the `LowerCtx` trait into its own impl `Lower`.

Reply:
Hmm, indeed; I want to try to keep things parameterized on the `Lower` trait (and not simplify to just the one impl) because it gives us flexibility later to e.g. be generic over multiple input IRs. This seems fine for now, I think.